python/lsst/ts/eas/m1m3_thermal.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This file is part of ts_eas.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+
+from dds import DDSException
+from lsst.ts import salobj
+
+SAL_TIMEOUT = 5.0  # SAL telemetry/command timeout
+REMOTE_STARTUP_TIME = 5.0  # Time for remotes to get set up
+SUMMARY_STATE_TIME = 5.0  # Wait time for a summary state change
+FAN_SLEEP_TIME = 30.0  # Time to wait after changing the fans
+VALVE_SLEEP_TIME = 60.0  # Time to wait after changing the valve
+DDS_RESTART_TIME = 60.0  # Time to wait after a DDS exception
+
+
+async def run_control_loop(domain: salobj.Domain, log: logging.Logger) -> None:
+    """Runs the control loop for the fans and the heaters.
+
+    Parameters
+    ----------
+    domain: salobj.Domain
+        The SALObj domain instance to use.
+
+    log: logging.Logger
+        A logger for log messages.
+    """
+    heaterdemand = [0] * 96
+    fandemand = [30] * 96
+
+    try:
+        m1m3ts = salobj.Remote(domain, "MTM1M3TS")
+        ess = salobj.Remote(domain, "ESS", index=112)
+
+        # Wait for remotes to get set up...
+        await asyncio.sleep(REMOTE_STARTUP_TIME)
+
+        mixing = await m1m3ts.tel_mixingValve.next(flush=True, timeout=SAL_TIMEOUT)
+        currentvalveposition = mixing.valvePosition
+        oldvalveposition = currentvalveposition
+
+        while True:
```

Comment:
Can the loop be moved into subroutine? Make this run, then run_loop running the true loop content.